### PR TITLE
test: FCM sample app with 2+ push event handlers

### PR DESCRIPTION
**Pull request by levibostian**
https://linear.app/customerio/issue/MBL-383/[bug]-customer-unable-to-handle-push-notification-event-on-ios-with

Follow-up to [this PR](https://github.com/customerio/customerio-ios/pull/751). This PR adds sample app testing. 

In order to test this bug fix in a sample app, the sample app needs 2+ 3rd party push event handlers installed. In our Flutter and RN sample apps, the 3rd party handlers are (1) FCM SDK and (2) the host iOS app. The native iOS FCM sample app uses SwiftUI which means that FCM swizzling is not enabled and therefore, we cannot rely on the FCM SDK being one of those 3rd party event handlers. In this PR, I created a new `UNUserNotificationCenterDelegate` class to get around this. 

How I tested this PR: 
* Run the FCM sample app. 
* Send a push notification to device while app in foreground. 
* View SDK logs. We expect to see logs say that the event was forwarded to the `AppDelegate` and to `AnotherPushEventHandler`. 

> Note: This is the same way I tested the Flutter, RN, and Expo sample apps. 

# Based on PR #750

## Full chain of PRs as of 2024-06-25

* PR #752: `levi/v3-multiple-push-handlers-sampleapp` ➔ `levi/v3-multiple-push-handlers-better-logs`
* PR #750: `levi/v3-multiple-push-handlers-better-logs` ➔ `levi/v3-multiple-push-handlers`
* PR #751: `levi/v3-multiple-push-handlers` ➔ `main`

### CODE REVIEW

**Review by Shahroz16 — APPROVED** [review]
changes LGTM

**Review by Shahroz16 — COMMENTED** [review]
This PR has `test` but it has the SDK changes which fixes the issue too. Is this intentional?

**Comment by levibostian** [comment]
@Shahroz16 sorry about the confusion. After merging the latest `main` into this PR, now you only see the sample app code changes. 

I merged the rest of the PR stack yesterday but did not merge this one. I think that's why this PR got out-of-date with it's target branch.

**Pull request merged**
